lightning/src/ln/channelmanager.rs

```diff
@@ -8263,6 +8263,53 @@ where
 						}
 					}
 				}
+
+				// Whether the downstream channel was closed or not, try to re-apply any payment
+				// preimages from it which may be needed in upstream channels for forwarded
+				// payments.
+				for (htlc_source, (htlc, preimage_opt)) in monitor.get_all_current_outbound_htlcs() {
```

Comment:
I haven't double checked this since the build is failing, but I think this would avoid a lot of indentation/be more idiomatic: 
[no_indentation.txt](https://github.com/lightningdevkit/rust-lightning/files/11455885/no_indentation.txt)